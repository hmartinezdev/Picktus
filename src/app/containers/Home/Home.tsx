import Background from '@components/Background';
import React, { Component } from 'react';

class Login extends Component {
  public render(): React.ReactElement<Login> {
    return (
      <Background>
        <div className="container">
          <style jsx>{`
            .container {
              width: 100%;
              height: 100vh;
              display: flex;
              align-items: center;
              flex-direction: column;
              padding: 2rem 0;
              box-sizing: border-box;
              overflow: auto;
            }
          `}</style>
        </div>
      </Background>
    );
  }
}

export default Login;
