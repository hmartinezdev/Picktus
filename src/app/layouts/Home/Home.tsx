import colors from '@constants/colors';
import ToolBar from '@containers/ToolBar';
import React, { Component } from 'react';

export interface IHomePropsType {
  status: string;
}

class Home extends Component<IHomePropsType> {
  public render(): React.ReactElement<Home> {
    const { status } = this.props;

    return (
      <div className={`container container--${status}`}>
        <div className="toolbarContainer">
          <ToolBar />
        </div>
        <style jsx>{`
          .container {
            display: flex;
            flex-direction: column;
            width: 100%;
            height: 100%;
            display: flex;
            align-items: center;
            flex-direction: column;
            box-sizing: border-box;
            overflow: auto;
            opacity: 1;
            align-items: flex-start;
            transition: opacity 300ms ease-out;
            background-color: ${colors.primary}cd;
          }

          .container--exiting,
          .container--exited {
            position: absolute;
            opacity: 0;
          }

          .container--entering {
            animation: opacity 300ms ease-in;
            z-index: 1;
          }

          .toolbarContainer {
            display: flex;
            align-items: flex-start;
            height: 100%;
            width: 12.5rem;
          }

          .toolbarAnimation {
            height: 100%;
            width: 100%;
          }

          @keyframes opacity {
            0% {
              opacity: 0;
            }

            100% {
              opacity: 1;
            }
          }
        `}</style>
      </div>
    );
  }
}

export default Home;
