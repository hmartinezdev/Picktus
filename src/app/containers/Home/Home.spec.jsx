import { shallow } from 'enzyme';
import React from 'react';
import Home from './Home';

let wrapper;

const setup = () =>
  shallow(
    <Home>
      <div />
    </Home>
  );

describe('<Home />', () => {
  test('should render properly', () => {
    wrapper = setup();
    expect(wrapper).toMatchSnapshot();
  });
});
