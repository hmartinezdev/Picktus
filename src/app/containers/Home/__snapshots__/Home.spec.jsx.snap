// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> should render properly 1`] = `
<Background>
  <div
    className="container"
  >
    <style
      jsx={true}
    >
      
            .container {
              width: 100%;
              height: 100vh;
              display: flex;
              align-items: center;
              flex-direction: column;
              padding: 2rem 0;
              box-sizing: border-box;
              overflow: auto;
            }
          
    </style>
  </div>
</Background>
`;
